### ISSUE TITLE
Provide Debian/Ubuntu compiled packages.

### ISSUE THREAD

**Issue by geertj** [aws/efs-utils#111]
Customers using Debian or Ubuntu want to use pre-built packages, like we have on Amazon Linux. Creating this issue to track interest.

**Comment by asokolsky4clari** [comment]
A package for Ubuntu 20.04 would be much appreciated.
Before such package becomes readily available, instructions on how to build it and make it available from an s3 bucket would help.

**Comment by briantist** [comment]
Any news on this one? `efs-utils` is the only way to use EFS with authentication and encryption in transit (some pretty critical features), and it's been a hard sell to make use of EFS when teams I work with can't rely on a trusted upstream repository.

**Comment by AnnoyingTechnology** [comment]
Any news on this 3 years old issue ?

**Comment by stevenolen** [comment]
We did a (very minimal and imperfect github actions-based release workflow in a fork, perhaps you could [use the release we made](https://github.com/rstudio/efs-utils/releases/tag/v2.0.1-1) or take the [action](https://github.com/rstudio/efs-utils/blob/master/.github/workflows/build-and-release.yaml) and fork/change as you need!

**Comment by marcus-sm8a** [comment]
Just wondering when the support for Debian 12 will be added :-)

**Comment by the-ar-chitect** [comment]
You can use this as **trustful** source if you are on Debian 13:
https://github.com/the-ar-chitect/aws-efs-utils-debian/releases/tag/amazon-efs-utils-debian13

It always tracks the latest version of aws efs-utils, automagically pulls and creates a package based on that.

**Why is this trustful?**
It uses a Github Action to **_transparently_** pull the repo and release a package, so no human intervention.
https://github.com/the-ar-chitect/aws-efs-utils-debian/blob/main/.github/workflows/build-efs-utils.yml
So you can trust it as long as you trust Github :)

And for making things easier, the latest version is always available with the same link:
https://github.com/the-ar-chitect/aws-efs-utils-debian/releases/download/amazon-efs-utils-debian13/amazon-efs-utils-latest.debian13_amd64.deb

**Comment by samuhale** [comment]
Supported distros now have vended rpms/debs available for use, see https://github.com/aws/efs-utils/blob/master/INSTALL.md for details. Ubuntu 20.04, 22.04, and 24.04 are supported. Debian is not currently available

**Comment by jsf9k** [comment]
See also #341, which fleshes out the argument for why this is needed.

**Comment by ArbitraryCritter** [comment]
@samuhale - It works great, well done.

For anyone here not willing to dissect their installer, this should get you started for Ubuntu:
`deb [signed-by=/usr/share/keyrings/efs-utils.gpg] https://amazon-efs-utils.aws.com/repo/deb/ubuntu/24.04 24.04 main`

You can retrieve the GPG key here:
https://amazon-efs-utils.aws.com/efs-utils-armored.gpg

Then install amazon-efs-utils:
`apt-get install amazon-efs-utils`

Or, use their installer:

`curl https://amazon-efs-utils.aws.com/efs-utils-installer.sh | sudo sh -s -- --install`

**Comment by UnwishingMoon** [comment]
Package for debian 13 would be appreciated.
Commenting to show interest in the feature request

**Comment by progiha** [comment]
> Package for debian 13 would be appreciated. Commenting to show interest in the feature request
Yeah, till then we go with the solution @the-ar-chitect provided above. That seems trustworthy and constantly auto-updated.
